### PR TITLE
Clean Architecture

### LINKED ISSUE #11: Introducing MVVM Architecture in the code
### Is your feature request related to a problem? Please describe.

To make a scalable app an Architecture plays a very important role. so to make this project a success we should add a architecture in it.

### Describe the solution you'd like

So I will restructure the existing code in MVVM format. So assign this issue to me.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #13: Adding Dependency injection in the app
### Is your feature request related to a problem? Please describe.

With the growing features in the app, there should be only single source of trust for the data. So for this purpose we should add Dagger Hilt in the project 

### Describe the solution you'd like

I will add Dagger hilt in the project so kindly assign it to me under GSSOC'23 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by essy16**
## Fixes Issue  

Closes #13
Closes  #11

## Changes proposed

--> I have added dependency Injection to the project
-->I have changed the architecture to Clean Architecture with MVVM

## Check List (Check all the applicable boxes) 

- [ x] My code follows the code style of this project.
- [ x] All new and existing tests passed.
- [ x] This PR does not contain plagiarized content.
- [ ]x The title of my pull request is a short description of the requested changes.

## Screenshots

Screen Shot 2023-06-13 at 5 17 18 PM

### CODE REVIEW

File: app/src/main/java/com/itssuryansh/taaveez/WebView.kt [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
 import android.view.View
-import android.webkit.WebResourceError
-import android.webkit.WebResourceRequest
 import android.webkit.WebViewClient
-import android.widget.Toast
 import androidx.annotation.RequiresApi
 import com.itssuryansh.taaveez.databinding.ActivityWebViewBinding
+import com.itssuryansh.taaveez.utilities.Constants
 
 class WebView : AppCompatActivity() {
```
Comment (Vaibhav2002) [inline app/src/main/java/com/itssuryansh/taaveez/WebView.kt:13]: Can you rename this to `WebViewActivity`?

File: app/src/main/java/com/itssuryansh/taaveez/domain/use_cases/GetAllNotesUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import android.util.Log
+import com.itssuryansh.taaveez.domain.model.NotesEntity
+import com.itssuryansh.taaveez.domain.repository.NotesRepository
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+
+  class GetAllNotesUseCase @Inject constructor (private val notesRepository: NotesRepository) {
+     operator fun invoke(): Flow<List<NotesEntity>> {
+         Log.d("GetAllNotesUseCase", "Fetching notes from repository...")
```
Comment (Vaibhav2002) [inline app/src/main/java/com/itssuryansh/taaveez/domain/use_cases/GetAllNotesUseCase.kt:11]: remove un-necessary logs

Comment (essy16) [reply app/src/main/java/com/itssuryansh/taaveez/domain/use_cases/GetAllNotesUseCase.kt:11]: Ok

File: app/src/main/java/com/itssuryansh/taaveez/WebView.kt (hunk 1 above)
Comment (essy16) [reply app/src/main/java/com/itssuryansh/taaveez/WebView.kt:13]: Uhm OK I'll do so

File: app/src/main/java/com/itssuryansh/taaveez/adapter/ItemAdapter.kt [hunk 3]
```diff
@@ -1,9 +1,10 @@
-package com.itssuryansh.taaveez
+package com.itssuryansh.taaveez.adapter
 
 import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.recyclerview.widget.RecyclerView
 import com.itssuryansh.taaveez.databinding.ItemPoemBinding
+import com.itssuryansh.taaveez.domain.model.NotesEntity
 import kotlin.collections.ArrayList
```
[the remark this answers is not in the document]
Comment (essy16) [reply app/src/main/java/com/itssuryansh/taaveez/adapter/ItemAdapter.kt:10]: I did my research and I found out it should be on the adapter package but I'll make the requested changes

File: app/src/main/java/com/itssuryansh/taaveez/presentation/NotesViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    fun getAllNotes() {
+        viewModelScope.launch {
+            val notes = getAllNotesUseCase.invoke().first()
+            _notesList.value = notes
+        }
+    }
+
+//    fun getNotesById(id: Int) {
```
Comment (Vaibhav2002) [inline app/src/main/java/com/itssuryansh/taaveez/presentation/NotesViewModel.kt:56]: remove this and reformat the file

File: app/src/main/java/com/itssuryansh/taaveez/presentation/ui/Notes.kt [hunk 5]
```diff
@@ -0,0 +1,529 @@
...
+import com.itssuryansh.taaveez.presentation.NotesViewModel
+import com.itssuryansh.taaveez.utilities.Constants
+import dagger.hilt.android.AndroidEntryPoint
+import jp.wasabeef.richeditor.RichEditor
+import kotlinx.coroutines.launch
+import java.text.SimpleDateFormat
+import java.util.*
+
+@AndroidEntryPoint
+class Notes : AppCompatActivity() {
```
Comment (Vaibhav2002) [inline app/src/main/java/com/itssuryansh/taaveez/presentation/ui/Notes.kt:44]: Can you please add the suffix Activity to all activity files, 
Eg. NotesActivity

**Review by Vaibhav2002 — CHANGES_REQUESTED** [review]
Great work man! Appreciate the effort.
Please make the required changes, then we're good to go.

File: app/src/main/java/com/itssuryansh/taaveez/adapter/ItemAdapter.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (Vaibhav2002) [reply app/src/main/java/com/itssuryansh/taaveez/adapter/ItemAdapter.kt:10]: Yup it should be inside Adapter package, but this Adapter package should be inside the Presentation Package

File: app/src/main/java/com/itssuryansh/taaveez/domain/model/NotesEntity.kt [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package com.itssuryansh.taaveez
+package com.itssuryansh.taaveez.domain.model
```
[the remark this answers is not in the document]
Comment (essy16) [reply app/src/main/java/com/itssuryansh/taaveez/domain/model/NotesEntity.kt:1]: What do you mean by this?

File: app/src/main/java/com/itssuryansh/taaveez/presentation/ui/Notes.kt (hunk 5 above)
Comment (essy16) [reply app/src/main/java/com/itssuryansh/taaveez/presentation/ui/Notes.kt:44]: Ok

File: app/src/main/java/com/itssuryansh/taaveez/presentation/NotesViewModel.kt (hunk 4 above)
Comment (essy16) [reply app/src/main/java/com/itssuryansh/taaveez/presentation/NotesViewModel.kt:56]: Remove what exactly?

File: app/src/main/java/com/itssuryansh/taaveez/domain/model/NotesEntity.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (Vaibhav2002) [reply app/src/main/java/com/itssuryansh/taaveez/domain/model/NotesEntity.kt:1]: This class is a database entity, which makes it a part of the Data Layer. 
Generally we create separate entities for separate layers of the app. Which means we have a separate class for Data Layer and a separate for Domain Level stuff. 

For example, a `NotesEntity` for data layer and a `Note` for domain layer, and mapper extension functions to map between the two.

Similarly if there is some Ui specific requirement in a Data class like such, we usually create a separate class for the Ui layer, So in this case we have a class for Data, Domain and Ui each, for the same Data Model

File: app/src/main/java/com/itssuryansh/taaveez/presentation/NotesViewModel.kt (hunk 4 above)
Comment (Vaibhav2002) [reply app/src/main/java/com/itssuryansh/taaveez/presentation/NotesViewModel.kt:56]: these comments

**Comment by Suryansh1720001** [comment]
Hello @essy16,
I believe these two issues have been assigned to Harsh, not you. If Harsh hasn't provided me with an update on his progress or allowed me to merge the code, I will proceed with merging it. 
By the way, I have some advice for you, always check if a particular issue has been assigned to you before working on it. Focus on the issues specifically assigned to you, and if you're interested in working on a different issue, you can comment on that specific issue to request its assignment to yourself, especially if the previous contributor is inactive.

**Comment by essy16** [comment]
Hello @Suryansh1720001  I'm sorry I didn't know ...I usually think that when there is an open issue any one is free to work on that issue but it's ok I've learnt my mistake and I won't be doing so again

**Comment by Vaibhav2002** [comment]
> Hello @Suryansh1720001 I'm sorry I didn't know ...I usually think that when there is an open issue any one is free to work on that issue but it's ok I've learnt my mistake and I won't be doing so again

You can start working on different issues for the time being. Refactoring the UI Layer might be a good start. 
Will keep this PR open, if the person assigned, does not complete or any other factor arises, we will continue with this PR.

**Comment by essy16** [comment]
Do you have suggestions of which issues I can work on?

**Comment by Vaibhav2002** [comment]
> Do you have suggestions of which issues I can work on?

As you can see, the Ui layer is quite messed up, everything is being done in Activity files, Refactoring that would be a great contribution. You can even add Navigation Components. 
Feel free to create an issue and explain your idea and process of implementing this. Will assign it to you

**Comment by essy16** [comment]
Hello @Suryansh1720001 Did Harsh create the PR?

**Pull request closed without merging**